tests/src/JIT/Regression/JitBlue/GitHub_17585/GitHub_17585.il

```diff
@@ -0,0 +1,151 @@
+// unmanaged, in this case it is JIT_TailCallHelperStub_ReturnAddress from tail call optimization.
+// To repro this issue on amd64/arm64 it also requires to reject fast tail call, for this puprose
+// `Interface::InterfaceMethodWithStackArgs` has several stack arguments.
+
+// The fix is to always generate explicit null check before calling VSD with tail call from tail call frame.
+
+
+.assembly extern System.Runtime
+{
+}
+.assembly extern System.Runtime.Extensions
+{
+}
+.assembly GitHub_17585
+{
+}
+
+
+// =============== CLASS MEMBERS DECLARATION ===================
+
+.class interface private abstract auto ansi Interface
+{
+  .method public hidebysig newslot abstract virtual 
+          instance int32  InterfaceMethodWithStackArgs(int32 a,
+                                                       int32 b,
+                                                       int32 c,
+                                                       int32 d,
+                                                       int32 e,
+                                                       int32 f,
+                                                       int32 g,
+                                                       int32 h,
+                                                       int32 i) cil managed
+  {
+  } // end of method Interface::InterfaceMethodWithStackArgs
+
+} // end of class Interface
+
+.class private auto ansi beforefieldinit InterfaceImplementation
+       extends [System.Runtime]System.Object
+       implements Interface
+{
+  .method public hidebysig newslot virtual final 
+          instance int32  InterfaceMethodWithStackArgs(int32 a,
+                                                       int32 b,
+                                                       int32 c,
+                                                       int32 d,
+                                                       int32 e,
+                                                       int32 f,
+                                                       int32 g,
+                                                       int32 h,
+                                                       int32 i) cil managed
+  {
+    // Code size       2 (0x2)
+    .maxstack  8
+    IL_0000:  ldc.i4.1
+    IL_0001:  ret
+  } // end of method InterfaceImplementation::InterfaceMethodWithStackArgs
+
+  .method public hidebysig specialname rtspecialname 
+          instance void  .ctor() cil managed
+  {
+    // Code size       7 (0x7)
+    .maxstack  8
+    IL_0000:  ldarg.0
+    IL_0001:  call       instance void [System.Runtime]System.Object::.ctor()
+    IL_0006:  ret
+  } // end of method InterfaceImplementation::.ctor
+
+} // end of class InterfaceImplementation
+
+.class private auto ansi beforefieldinit Test
+       extends [System.Runtime]System.Object
+{
+  .method public hidebysig static int32  CallInterfaceMethod(class Interface iface) cil managed
```

Comment:
Do you also need noinline on this to ensure that the method won't get inlined?

When you have the pile of boxing on it in previous iterations, it was big enough for inliner to inline. But now that it is just a wrapper, it may be getting close to the inlining treas-hold.

Reply:
It will work even if it is inlined, but I think it is good idea to make it more solid.
Fixed.